app/src/main/java/me/ccrama/redditslide/Activities/Submit.java

```diff
@@ -585,12 +585,16 @@ public File createFile(Uri uri, @NonNull Context context) {
             String type = resolver.getType(uri);
             String extension;
 
-            if ("image/png".equals(type)) {
-                extension = ".gif";
-            } else if ("image/png".equals(type)) {
-                extension = ".png";
-            } else {
-                extension = ".jpg";
+            switch (type) {

```

Comment:
this switch statement (used in three places) was the bug fix.
`image/png` was being caught twice. The first catch set the extension as a `.gif` erroneously causing `pngs` to save as `gifs` and `gifs` to end up as `jpgs` or just never save. Depending how this was used.